GTG/tests/test_backend_bugzilla.py

```diff
@@ -0,0 +1,354 @@
+    '''
+    Get the raw content text without XML tags and task tags
+
+    @param content: the content string returned by Task.get_text method.
+    @return: the cleaned content
+    '''
+    xmldoc = parseString(content)
+    root = xmldoc.documentElement
+    # For this testing, the last element with type TextNode should be the one
+    # I want.
+    result = root.childNodes[-1].nodeValue
+    return result.strip()
+
+
+def test_suite():
+    return unittest.TestLoader().loadTestsFromName(__name__)
+
+
+##############################################################################
+# Fake objects and control methods for testing
+#
+
+
+class FakeGObject(object):
+    '''Fake GObject to avoid to wait real gobject's event loop'''
+
+    def idle_add(self, func, *args, **kwargs):
+        return func(*args, **kwargs)
+
+
+class FakeLocalhostBugzillaService(BugzillaService):
+    '''Fake Bugzilla service running on localhost for testing'''
+
+    name = 'Fake local Bugzilla Service'
+
+
+class FakeBug(Bug):
+    '''Fake Bug for testing'''
+
+
+STOCK_FAKE_BUGS = {
+    1: {
+        'id': 1,
+        'summary': '[RFE] add fake bugzilla service for testing',
+        'component': ['Application', 'Database']
+    },
+    2: {
+        'id': 2,
+        'summary': '[FeatureRequest] bugzilla service should be '
+                   'configurable',
+        'component': ['Installer', 'GUI', 'DBus']
+    }
+}
+
+
+class FakeBugzillaWSBug(object):
+    '''Fake Bugzilla Webservice Bug'''
+
+    def get(self, options):
+        '''
+        Get bugs
+
+        @param options: a dictionary object, containing ids at least.
+        @return: a list of bugs, each of those is a dictionary object.
+        '''
+        bugs = []
+        ids = options.get('ids', ())
+        for bug_id in ids:
+            bugs.append(STOCK_FAKE_BUGS.get(int(bug_id)))
+        return {
+            'bugs': bugs,
+            'faults': []
+        }
+
+
+class FakeBugzillaService(object):
+    '''Fake Bugzilla servive'''
+
+    def __init__(self):
+        self.Bug = FakeBugzillaWSBug()
+
+
+def run_fake_bugzilla_service():
+    '''
+    Run a fake Bugzilla service in a separate process
+
+    @return: Popen object representing the process running fake Bugzilla
+             service.
+    '''
+    proc = Process(target=run_xmlrpc_server)
+    proc.start()
+    return proc
+
+
+class RequestHandler(SimpleXMLRPCRequestHandler):
+    rpc_paths = ('/xmlrpc.cgi',)
+
+
+def run_xmlrpc_server():
+    '''
+    Run the XMLRPC server hosting fake Bugzilla service
+
+    This method will be invoked in the separate process launched for running
+    the fake Bugzilla service.
+    '''
+
+    server = SimpleXMLRPCServer((XMLRPC_HOST, XMLRPC_PORT),
+                                logRequests=False,
+                                requestHandler=RequestHandler)
+    server.register_instance(FakeBugzillaService(), allow_dotted_names=True)
+    server.serve_forever()
```

Comment:
I don't like this unit test. Ideally you should split your code in that sense that it should be easy to mock the returned objects from the remote service like Bugzilla and test the rest of the object without spinning up a new XML RPC server. That would be slow & cumbersome and it wouldn't provide much benefit.

Have a look at [tests/core/test_config.py](https://github.com/getting-things-gnome/gtg/blob/master/tests/core/test_config.py) that does something similar: tests the loading up configuration from disk without actually touching the disk.